k9mail/src/main/java/com/fsck/k9/activity/compose/CryptoSettingsDialog.java

```diff
@@ -104,21 +118,41 @@ public void onCryptoStatusSelected(int status) {
         }
         updateView(true);
 
+        OnCryptoModeChangedListener activity = getCryptoModeChangeListenerActivity();
+        if (activity == null) {
+            return;
+        }
+
+        activity.onCryptoModeChanged(currentMode);
+    }
+
+    @Override
+    public void onCheckedChanged(CompoundButton compoundButton, boolean switchState) {
+        OnCryptoModeChangedListener activity = getCryptoModeChangeListenerActivity();
+        if (activity == null) {
+            return;
+        }
+
+        activity.onCryptoCompatChanged(switchState);
+    }
+
+    @Nullable
+    private OnCryptoModeChangedListener getCryptoModeChangeListenerActivity() {
         Activity activity = getActivity();
         if (activity == null) {
             // is this supposed to happen?
```

Comment:
both times this is called, it is triggered by a user interaction. That can only happen when the activity is available, so this *should* never happen. Android activity lifecycle does wacky things though, so I wouldn't throw an illegal state here...